### PR TITLE
Adding New CorsFilter from api-security-java to the web service 

### PR DESCRIPTION
**Pull request by hselvakumar**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -108,6 +120,10 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
+        </dependency>
```
Comment (bcullerton) [inline pom.xml:126]: Is this needed with the import above?

File: pom.xml [hunk 2]
```diff
@@ -27,13 +27,15 @@
...
         <commons.io.version>2.16.1</commons.io.version>
+        <api-security-java.version>2.0.5</api-security-java.version>
 
         <!-- internal dependencies -->
         <structured-logging.version>3.0.4</structured-logging.version>
         <sdk-manager-java.version>3.0.9</sdk-manager-java.version>
         <api-sdk-manager-java-library.version>3.0.5</api-sdk-manager-java-library.version>
         <private-api-sdk-java.version>4.0.99</private-api-sdk-java.version>
         <api-helper-java-library.version>3.0.1</api-helper-java-library.version>
+        <api-security-java.version>2.0.5</api-security-java.version>
```
Comment (NeilShadrach) [inline pom.xml:38]: Duplicates L30

**Review by NeilShadrach — CHANGES_REQUESTED** [review]

File: pom.xml [hunk 3]
```diff
@@ -108,6 +120,10 @@
             <groupId>org.springframework.boot</groupId>
             <artifactId>spring-boot-starter-web</artifactId>
         </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-security</artifactId>
```
Comment (NeilShadrach) [inline pom.xml:125]: Duplicates L107

**Review by NeilShadrach — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 2 above)
Comment (hselvakumar) [reply pom.xml:38]: removed now

File: pom.xml (hunk 1 above)
Comment (hselvakumar) [reply pom.xml:126]: removed duplicates

File: pom.xml (hunk 3 above)
Comment (hselvakumar) [reply pom.xml:125]: removed duplicates

File: src/main/java/uk/gov/companieshouse/company_appointments/config/Config.java [hunk 4]
```diff
@@ -28,14 +28,16 @@
...
-
-    @Autowired
-    private AuthenticationInterceptor authenticationInterceptor;
+    private final RequestLoggingInterceptor loggingInterceptor;
+    private final AuthenticationInterceptor authenticationInterceptor;
+    private final FullRecordAuthenticationInterceptor fullRecordAuthenticationInterceptor;
 
     @Autowired
-    private FullRecordAuthenticationInterceptor fullRecordAuthenticationInterceptor;
+    public Config (RequestLoggingInterceptor loggingInterceptor, AuthenticationInterceptor authenticationInterceptor, FullRecordAuthenticationInterceptor fullRecordAuthenticationInterceptor){
```
Comment (NeilShadrach) [inline src/main/java/uk/gov/companieshouse/company_appointments/config/Config.java:36]: Someone will have to explain this one to me - I see it all over the place. I thought the point of @Autowired on the private variables was that the constructor would then be generated automatically

**Review by bcullerton — APPROVED** [review]

File: src/main/java/uk/gov/companieshouse/company_appointments/config/CorsConfiguration.java [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.security.config.annotation.web.configurers.AbstractHttpConfigurer;
+import org.springframework.security.web.SecurityFilterChain;
+import org.springframework.security.web.csrf.CsrfFilter;
+import uk.gov.companieshouse.api.filter.CustomCorsFilter;
+
+import java.util.List;
+
+@Configuration
+public class CorsConfiguration {
+@Bean
```
Comment (NeilShadrach) [inline src/main/java/uk/gov/companieshouse/company_appointments/config/CorsConfiguration.java:17]: Alignment

**Pull request merged**